treescript/src/treescript/merges.py

```diff
@@ -0,0 +1,140 @@
+            new_contents += f"{line}\n"
+    new_contents += "Merge day clobber"
+    with open(clobber_file, "w") as f:
+        f.write(new_contents)
+
+
+async def apply_rebranding(config, repo_path, merge_config):
+    """Apply changes to repo required for merge/rebranding."""
+    log.info("Rebranding %s to %s", merge_config.get("from_branch"), merge_config.get("to_branch"))
+    if merge_config.get("version_files"):
+        current_version = get_version("browser/config/version.txt", repo_path)
+        next_version = f"{current_version.major_number}.{current_version.minor_number}"
+
+        await do_bump_version(config, repo_path, merge_config["version_files"], next_version)
+    if merge_config.get("version_files_suffix"):
+        current_version = get_version("browser/config/version.txt", repo_path)
+        next_version = f"{current_version.major_number}.{current_version.minor_number}{merge_config.get('version_suffix')}"
+
+        await do_bump_version(config, repo_path, merge_config["version_files_suffix"], next_version)
+
+    for f in merge_config.get("copy_files", list()):
+        shutil.copyfile(os.path.join(repo_path, f[0]), os.path.join(repo_path, f[1]))
+
+    for f, from_, to in merge_config.get("replacements", list()):
+        replace(os.path.join(repo_path, f), from_, to)
+
+    touch_clobber_file(repo_path)
+
+
+# do_merge {{{1
+async def do_merge(config, task, repo_path):
+    """Perform a merge day operation.
+
+    This function takes its inputs from task's payload.
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+        repo_path (str): the source directory
+
+    Raises:
+        TaskverificationError: from get_merge_config if the payload is invalid.
+
+    Returns:
+        list: A list of the branches that need pushing, and the corresponding revision.
+              This is unlike other actions as the list of outgoing changes is
+              not related to the number of commands we've performed, but we do need
+              to know which branches to push.
+    """
+    merge_config = get_merge_config(task)
+
+    from_branch = merge_config.get("from_branch")
+    to_branch = merge_config.get("to_branch")
+
+    await run_hg_command(config, "pull", "https://hg.mozilla.org/mozilla-unified", repo_path=repo_path)
+
+    await run_hg_command(config, "up", "-C", from_branch, repo_path=repo_path)
+
+    base_from_rev = await get_revision(config, repo_path, branch=from_branch)
+
+    base_tag = merge_config["base_tag"].format(major_version=get_version("browser/config/version.txt", repo_path).major_number)
+
+    tag_message = f"No bug - tagging {os.path.basename(repo_path)} with {base_tag} a=release DONTBUILD CLOSED TREE"
+    await run_hg_command(config, "tag", "-m", '"{}"'.format(tag_message), "-r", base_from_rev, "-f", base_tag, repo_path=repo_path)
+
+    tagged_from_rev = await get_revision(config, repo_path, branch=from_branch)
```

Comment:
This will need to be `branch="."` (or based on my other comment `revset="."`); branch names we are using here are static. `.` refers to the current commit.